### ISSUE TITLE
Add reference line

### ISSUE THREAD

**Issue by hlehmann** [uber/react-vis#371]
In the same way it's possible to had hint it would be nice to be able to had reference lines.

I will take example: http://recharts.org/#/en-US/api/ReferenceLine

**Comment by samhogg** [comment]
@hlehmann Probably not the most intuitive naming for you but this is already achievable using the [static crosshair](http://uber.github.io/react-vis/#/examples/charts/showcase) (look towards the bottom of that page)

**Comment by mcnuttandrew** [comment]
That's a great point @samhogg! I think the right move might be for someone to write up a more explicit guide on how to create that type annotation. 

I think this ticket also raises some interesting questions about how we address annotation in the library. Right now it is limited to Crosshairs and Tooltips, but it could be so much more (http://d3-annotation.susielu.com/)

**Comment by samhogg** [comment]
Agreed, that library is fantastic! Wonder if we should have a general "annotations" issue to track related stuff

Have you guys got any sort of roadmap in mind for `react-vis`, @mcnuttandrew?

**Comment by pcmaffey** [comment]
I tried using a static crosshair for a 'reference line' on a time series plot, but it appears the Crosshair component can't handle dates?
`<Crosshair
          values={[{x: data[0].x, y: avg}, {x: data[data.length - 1].x, y: avg}]}
  />`

Errors: 

> Uncaught Error: Objects are not valid as a React child (found: Sat Jun 24 2017 18:47:21 GMT-0600 (MDT))

**Comment by mcnuttandrew** [comment]
Hey @pcmaffey 

I just checked using crosshairs with with time data, and it appears to work fine. How are you formatting your dates? Check out the time-chart for an example of a good way to handle that type of thing.

As for using the DecorativeAxis, it sounds like you probably didn't give it all of the arguments? axisStart and axisEnd are object of the format {x, y}
